### PR TITLE
TSK integration and example tables

### PR DESCRIPTION
**Pull request by theopolis**
Add a `device_file` and `device_partitions` table, backed by Sleuthkit APIs for reading RAW (or otherwise, but not tested) disk images.

On OS X one may:

```sql
select * from device_file where device = '/dev/rdisk0' and partition = '4'
```

To read contents of unmounted partitions,

### CODE REVIEW

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

File: osquery/tables/forensic/sleuthkit.cpp [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+class DeviceHelper : private boost::noncopyable {
+ public:
+  DeviceHelper(const std::string& device_path)
+      : image_(std::make_shared<TskImgInfo>()),
+        volume_(std::make_shared<TskVsInfo>()),
+        device_path_(device_path) {}
+
+  /// Volume partition iterator.
+  void forEachPartition(
```
Comment (marpaia) [inline osquery/tables/forensic/sleuthkit.cpp]: When we use this pattern in `osquery::Config`, we don't specify the kind of iteration. For example, there's `Config::scheduledQueries`, not `Config::mapScheduledQueries` or `Config::forEachScheduledQueries`. Minor nit would be to change this to `DeviceHelper::partitions`

Comment (theopolis) [inline osquery/tables/forensic/sleuthkit.cpp]: Good point!

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Pull request merged**